Sources/ApodiniNetworking/HTTP/Headers.swift

```diff
@@ -0,0 +1,498 @@
+    subscript(name: String) -> [String] { get }
+}
+
+
+extension NIOHPACK.HPACKHeaders: __ANNIOHTTPHeadersType {}
+
+
+extension NIOHTTP1.HTTPHeaders: __ANNIOHTTPHeadersType {
+    public mutating func add(name: String, value: String, indexing: HPACKIndexing) {
+        add(name: name, value: value)
+    }
+    
+    public mutating func replaceOrAdd(name: String, value: String, indexing: HPACKIndexing) {
+        replaceOrAdd(name: name, value: value)
+    }
+}
+
+
+/// An untyped HTTP/1 or HTTP/2 header key.
+public class AnyHTTPHeaderName: Equatable {
+    public let rawValue: String
+    internal init(_ value: String) {
+        self.rawValue = value
+    }
+    
+    public static func == (lhs: AnyHTTPHeaderName, rhs: AnyHTTPHeaderName) -> Bool {
+        ObjectIdentifier(lhs) == ObjectIdentifier(rhs)
+    }
+}
+
+
+/// A typed HTTP (1 or 2) header name, i.e. the key in a header field.
+/// - Note: The standard differentiates between single-value and multi-value header fields
+///         (i.e. headers which can have only one value, and also appear only once; and headers which can have multiple values and also may appear multiple times).
+///         You can control whether a header name defined using this API is a single- or multi-value header by wrapping the header type in an `Array`.
+///         For example: `let date = HeaderName<Date>("Date")` defines a single-value header (the Date header may only appear once, and can only contain one value).
+///         On the other hand, `let contentEncoding = HeaderName<[HTTPContentEncoding]>("Content-Encoding")` would define a multi-value header, which can contain one or more values, and may also appear multiple times itself.
+/// See also: https://datatracker.ietf.org/doc/html/rfc7230#section-3.2
+public class HTTPHeaderName<T: HTTPHeaderFieldValueCodable>: AnyHTTPHeaderName {
+    override public init(_ value: String) {
+        super.init(value)
+    }
+}
+
+
+extension __ANNIOHTTPHeadersType {
+    /// Creates a new headers struct, giving the caller the opportunity to initialise it via the closure.
+    /// - Note: The reason this initialiser exists is to offer a type-safe way of declaring immutable headers objects.
+    ///         The underlying problem here is that Swift doesn't support variadic generics, meaning that (since the type-safe header
+    ///         names are implemented via generics) you wouldn't be able to pass more than one generic key-value pair to the initialiser.
+    ///         This initialiser wors around that by giving the caller the ability to access a mutable version of the headers
+    ///         struct (which can be modified using the type-safe API), while still allowing the caller to store the resulting value into an immutable object.
+    ///         Sinice the block is non-escaping, immediately evaluated, and evaluated only once, this is semantically equivalent to declaring a mutable headers object and modifying that.
```

Comment:
```suggestion
    /// - Note: The reason this initializer exists is to offer a type-safe way of declaring immutable headers objects.
    ///         The underlying problem here is that Swift doesn't support variadic generics, meaning that (since the type-safe header
    ///         names are implemented via generics) you wouldn't be able to pass more than one generic key-value pair to the initializer.
    ///         This initializer works around that by giving the caller the ability to access a mutable version of the headers
    ///         struct (which can be modified using the type-safe API), while still allowing the caller to store the resulting value into an immutable object.
    ///         Since the block is non-escaping, immediately evaluated, and evaluated only once, this is semantically equivalent to declaring a mutable headers object and modifying that.
```